Yosemite/Yosemite/Stores/ProductAttributeTermStore.swift

```diff
@@ -101,6 +103,24 @@ private extension ProductAttributeTermStore {
             }
         }
     }
+
+    /// Creates and stores a new product attribute term and links it to it's parent product attribute.
+    ///
+    func createProductAttributeTerm(siteID: Int64,
+                                    attributeID: Int64,
+                                    name: String,
+                                    onCompletion: @escaping (Result<ProductAttributeTerm, Error>) -> Void) {
+        remote.createProductAttributeTerm(for: siteID, attributeID: attributeID, name: name) { [weak self] result in
+            switch result {
+            case let .success(term):
+                self?.upsertStoredProductAttributeTermsInBackground([term], siteID: siteID, attributeID: attributeID) {
```

Comment:
nit: maybe we can guard on `self` at the beginning of the closure so that when it's `nil`, it doesn't only go through the `failure` case?

Reply:
Nice catch!